src/main/java/com/userservice/config/UserServiceInterceptor.java

```diff
@@ -0,0 +1,62 @@
+package com.userservice.config;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.HandlerInterceptor;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@Slf4j
+@Component
+public class UserServiceInterceptor implements HandlerInterceptor {
+
+    private TokenValidationUtil jwtTokenUtil;
+
+    public UserServiceInterceptor(TokenValidationUtil jwtTokenUtil) {
+        this.jwtTokenUtil = jwtTokenUtil;
+    }
+
+    /**
+     * This function performs various operations on the intercepted request then allow that request to the original
+     * server after some validation checks.
+     *
+     * @param request
+     * @param response
+     * @param handler
+     * @return boolean
+     * @throws Exception
+     */
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler)
+            throws Exception {
+        log.info("Entering UserServiceInterceptor.preHandle with parameters request {}, response {} and handler {}",
+                request, response, handler);
+        boolean applyThrough = false;
+        final String requestTokenHeader = request.getHeader("Authorization");
+        String userName = null;
+        String jwtToken = null;
+        String requestParams = request.getQueryString();
+
+        if (requestParams != null) {
+            if (requestParams.contains("userName") && !requestParams.contains("password"))
+                applyThrough = true;
```

Comment:
if the username is not null, and password is null.. Then why dont you `return true` from here itself

Reply:
Done.